### PR TITLE
Merge corrected LuaLS annotation code

### PR DESCRIPTION
**Pull request by bjcscat**
## Ticket

Resolves requests made in #46 

## Changes

Changed the luals-annotations package to be in line with the changes requested.

## Impact

These changes should allow for the luals annotations to be merged into master

## Testing

Tests remain as they are in #46

### CODE REVIEW

**Comment by Derpius** [comment]
Thanks for this! I’ve had a quick scan on the way to work and looks good. I’ll do a proper review this weekend though (also planning to smash through the backlog a bit)

The commitlint pipeline is failing due to your commit not being prefixed, but im planning to replace conv commits with a changelog doc for the automated releases this weekend anyway

**Comment by bjcscat** [comment]
Did a force-push with simplified commit history and a proper prefix.

**Comment by bjcscat** [comment]
Additionally, the openresty CI job seems to be broken

File: packages/luals-annotations/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+classDocs.forEach(generateClass);
+functionDocs.forEach((func) => document.addFunction(func));
+generateMatcherClass({ inverse: false });
+generateMatcherClass({ inverse: true });
+
+// This environment variable is set by NPM when you pass --debug to itw
+
+if (process.env["npm_config_debug"]) {
```
Comment (Derpius) [inline packages/luals-annotations/src/index.ts]: nit:
```suggestion
// This environment variable is set by NPM when you pass --debug to it
if (process.env["npm_config_debug"]) {
```

**Review by Derpius — APPROVED** [review]
I think this is good enough now (although I've left one nit on a typo). There's still a lot of tech debt from @yogwoggf's original implementation that would ideally be fixed, but given it just needs to generate annotations and doesn't affect any critical part of the codebase then I think merging now is the pragmatic solution

Taking a look at the openresty pipeline first, then I'll do a quick QA

**Pull request merged**